packages/shorebird_cli/lib/src/extensions/string.dart

```diff
@@ -1,4 +1,21 @@
+import 'dart:io';
+
+import 'package:shorebird_cli/src/code_signer.dart';
+
 extension NullOrEmtpy on String? {
   /// Returns `true` if this string is null or empty.
   bool get isNullOrEmpty => this == null || this!.isEmpty;
 }
+
+extension SignString on (String, File?) {
+  String? get signature {
+    final privateKeyFile = this.$2;
+
+    return privateKeyFile != null
+        ? codeSigner.sign(
+            message: this.$1,
+            privateKeyPemFile: privateKeyFile,
+          )
+        : null;
+  }
+}
```

Comment:
I don't love this as an extension on a record with two unnamed fields. This isn't saving us much code, so it might be better to inline this.

Reply:
I agree. Also it's weird that signature returns a nullable string and doesn't do anything if a null file is passed. I think it'd be better have signature require a private key and always return a signature and we can conditionally call `sign` instead.